features/rule_creation_protocol/version_check.py

```diff
@@ -0,0 +1,29 @@
+import subprocess
+import re
+from .errors import ProtocolError
+
+def get_changed_feature_files(target_branch):
+    changed_files = subprocess.getoutput(f"git diff --name-only {target_branch}").split("\n")
+    feature_files = [f for f in changed_files if f.endswith(".feature")]
+    return feature_files
+
+def check_version_bump(file):
+    # Compare the file from the current branch to the target_branch
+    diff_output = subprocess.getoutput(f"git diff {file}")
+    versions = re.findall(r"@version(\d+)", diff_output)
+    versions = [int(v) for v in versions]
+    if not versions:
+        raise ProtocolError(
+            value = None,
+            message = "When changing a feature file, the version number must be bumped"
+        )
+    return versions[1] > versions[0]  # Check version bump
```

Comment:
Ah you're using the diff output. I can't think of a way where it would break. I have used https://pypi.org/project/unidiff/ in the past to more properly parse the diff output, but wouldn't necessarily recommend. Also had some issues.

I would add a check though about `if len(versions) > 2`. I guess that could happen if you add multiple version tags? Or is that already checked elsewhere. Minor thing.

I wonder if we should check `return versions[1]  == versions[0] + 1`?